src/main/java/games/strategy/engine/framework/startup/ui/GameSelectorPanel.java

```diff
@@ -370,25 +371,41 @@ private void selectGameFile(final boolean saved) {
       if (file == null || !file.exists()) {
         return;
       }
-      model.load(file, this);
-      setOriginalPropertiesMap(model.getGameData());
+      try {
+        BackgroundTaskRunner.runInBackgroundAndReturn("Loading savegame...", () -> {
+          model.load(file, this);
+          setOriginalPropertiesMap(model.getGameData());
+          return null;
+        });
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+      }
     } else {
-      final NewGameChooserEntry entry =
-          NewGameChooser.chooseGame(JOptionPane.getFrameForComponent(this), model.getGameName());
+      final GameChooserEntry entry =
+          GameChooser.chooseGame(JOptionPane.getFrameForComponent(this), model.getGameName());
       if (entry != null) {
-        if (!entry.isGameDataLoaded()) {
-          try {
-            entry.fullyParseGameData();
-          } catch (final GameParseException e) {
-            entry.delayParseGameData();
-            NewGameChooser.getNewGameChooserModel().removeEntry(entry);
-            return;
-          }
+        try {
+          BackgroundTaskRunner.runInBackgroundAndReturn("Loading map...", () -> {
+            synchronized (entry) {
```

Comment:
Not sure this is doing anything useful in terms of thread safety.  The original concern here was publishing values written by one thread to other threads that read it, not necessarily concurrent access (which doesn't appear possible here).

However, after we discussed in your other PR what `Future#get()` is doing behind the scenes in terms of creating a memory barrier, I'm going to assume `SwingWorker#get()` makes the same guarantee (in fact, it looks like `SwingWorker` uses `Future` internally).  So, I think it's safe to assume that values written by the background thread will be published correctly to the EDT once `runInBackgroundAndReturn()` is complete.

TL;DR: Remove the `synchronized` block; your original code is ok. :smile: